Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
-    case 9,
-        % Exit image display:
-        %         fprintf('exit_image_display.\n');
-        %         fprintf('AVG FPS = %f Hz\n', drawcount / (GetSecs - lastImageTime));
+        
+    case 9  % Exit Image Display
         clearScreen=1;
         ineyeimagemodedisplay=0;
         drawInstructions=1;
         needsupdate = 1;
-    case 10,
-        % Erase current calibration target:
-        %         fprintf('erase_cal_target.\n');
+        
+    case 10 % Erase Cal Target
         calxy = [];
+        if ~isempty(eyelinkanimationtarget)
+            eyelinkanimationtarget.calxy=calxy;
+        end
         clearScreen=1;
         needsupdate = 1;
-    case 11,
-        %               fprintf('exit_cal_display.\n');
-        %         fprintf('AVG FPS = %f Hz\n', drawcount / (GetSecs - lastImageTime));
-        clearScreen=1;
-        %       drawInstructions=1;
-        needsupdate = 1;
-    case 12,
-        % New calibration target sound:
-        %         fprintf('cal_target_beep_hook.\n');
-        EyelinkMakeSound(el, 'cal_target_beep');
-    case 13,
-        % New drift correction target sound:
-        %         fprintf('dc_target_beep_hook.\n');
-        EyelinkMakeSound(el, 'drift_correction_target_beep');
-    case 14,
-        % Calibration done sound:
-        errc = callArgs(2);
-        %         fprintf('cal_done_beep_hook: %i\n', errc);
-        if errc > 0
-            % Calibration failed:
-            EyelinkMakeSound(el, 'calibration_failed_beep');
-        else
-            % Calibration success:
-            EyelinkMakeSound(el, 'calibration_success_beep');
+        
+        if strcmpi(el.calTargetType, 'video') && ~isempty(eyelinkanimationtarget)
+             texkill=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTexClose);
+             Screen('PlayMovie', eyelinkanimationtarget.movie, 0, el.calAnimationLoopParam);
+             if texkill > 0
+                 Screen('Close', texkill);
+             end
+        end
+        
+    case 11 % Exit Cal Display
+        calxy = [];
+        if(~isempty(eyelinkanimationtarget)	)
+            eyelinkanimationtarget.calxy=calxy;
         end
-    case 15,
-        % Drift correction done sound:
-        errc = callArgs(2);
-        %         fprintf('dc_done_beep_hook: %i\n', errc);
-        if errc > 0
-            % Drift correction failed:
-            EyelinkMakeSound(el, 'drift_correction_failed_beep');
+        
+        if inDrift
+            inDrift = 0;
+            drawInstructions = 0;
         else
-            % Drift correction success:
-            EyelinkMakeSound(el, 'drift_correction_success_beep');
+            drawInstructions = 1;
+        end
+        
+        clearScreen=1;
+        needsupdate=1;
+        if strcmpi(el.calTargetType, 'video') && ~isempty(eyelinkanimationtarget)
+            texkill=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTexClose);
```

Comment:
ditto

Reply:
understood this can likely be done away with. Wish to leave it in for now, and save revisions to this part of the code -- that is distributed b/w DoCameraSetup, DoDriftCorrection, and the callback handlers /w shared structs, globals, etc. -- until a later time to include more thorough testing of the changes.